addons/mail/models/mail_followers.py

```diff
@@ -145,3 +70,166 @@ def unlink(self):
         ('mail_followers_res_channel_res_model_id_uniq', 'unique(res_model,res_id,channel_id)', 'Error, a channel cannot follow twice the same object.'),
         ('partner_xor_channel', 'CHECK((partner_id IS NULL) != (channel_id IS NULL))', 'Error: A follower must be either a partner or a channel (but not both).')
     ]
+
+    # --------------------------------------------------
+    # Private tools methods to fetch followers data
+    # --------------------------------------------------
+
+    def _get_followers_data(self, res_model, res_ids, pids, cids):
+        where_params = (res_model, res_ids, pids, cids)
+        query = """
+SELECT fol.id, fol.res_id, fol.partner_id, fol.channel_id, array_agg(subtype.id)
+FROM mail_followers fol
+LEFT JOIN mail_followers_mail_message_subtype_rel fol_rel ON fol_rel.mail_followers_id = fol.id
+LEFT JOIN mail_message_subtype subtype ON subtype.id = fol_rel.mail_message_subtype_id
+WHERE fol.res_model = %s AND fol.res_id = ANY(%s) AND
+    (fol.partner_id = ANY(%s) OR fol.channel_id = ANY(%s))
+GROUP BY fol.id"""
+        self.env.cr.execute(query, where_params)
+        return self.env.cr.fetchall()
+
+    def _get_doc_follower_data(self, doc_data):
```

Comment:
This method is private, therefore it may accept a list of recordsets instead of a list of (model, ids).
However, I don't know whether this is more convenient for the callers of the methods.

Reply:
Fine, that answers the question.